### PR TITLE
Check working tests

### PR DESCRIPTION
**Pull request by VoronovaIntern**

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
@VoronovaIntern please rebase, also list which tests still disabled (and why)

**Comment by VoronovaIntern** [comment]
tests are falling
CPP dependences cannot be built in human_pose_estimation_3d_demo and speech_recognition_deepspeech_demo  
Python:
- [ ] human_pose_estimation_3d_demo : ImportError: cannot import name 'extract_poses' from 'pose_extractor' (unknown location)
- [ ] speech_recognition_deepspeech_demo : No module named 'ctcdecode_numpy'
- [ ] bert_named_entity_recognition_demo : IndexError: list index out of range
- [ ]  image_retrieval_demo : Need to Put image_retrieval_arg import back

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by VoronovaIntern** [comment]
added several fails:

- [ ] instance_segmentation_demo : division by zero
- [ ]  social_distance_demo person-reidentification-retail-0277 : fails with exit code -6
- [ ]  object_detection_demo face-detection-0205:  fails with exit code -11

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
failed at person-reidentification-retail-0287, but not reproduced locally (at least on video)

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by VoronovaIntern** [comment]
Now commented python demos which cannot find modules

- [ ] speech_recognition_deepspeech_demo (No module named 'ctcdecode_numpy')
- [ ] image_retrieval_demo (TODO: Put image_retrieval_arg import back) 
- [ ] human_pose_estimation_3d_demo (cannot import name 'extract_poses' from 'pose_extractor' (unknown location))

**Comment by vladimir-dudnik** [comment]
@nignatovsky it seems to be some CI env issue when python demos can't find modules. Do we add them to PYTHONPATH as it described in these demo description?

**Comment by anzhella-pankratova** [comment]
@VoronovaIntern you have some not related to tests changes in the PR. Could you please remove it?

**Comment by VoronovaIntern** [comment]
I ran tests locally and only demo with person-reidentification-retail-0287 and 0288 on version 612, everything works

**Comment by Wovchena** [comment]
... after you resolve the conflict :)

**Comment by vladimir-dudnik** [comment]
@VoronovaIntern please rebase

**Comment by VoronovaIntern** [comment]
Jenkins please retry a build

File: demos/tests/cases.py [hunk 1]
```diff
@@ -310,33 +309,77 @@ def single_option_cases(key, *args):
...
+            '-m_lm': ModelArg('facial-landmarks-35-adas-0002'),
+            '-m_es': ModelArg('open-closed-eye-0001'),
+        }),
+        single_option_cases(
+            '-m_fd',
+            ModelArg('face-detection-adas-0001'),
+            ModelArg('face-detection-retail-0004')),
+    )),
+
+    CppDemo(name='gaze_estimation_demo', implementation='cpp_gapi',
```
Comment (anzhella-pankratova) [inline demos/tests/cases.py:338]: The repetition, need to remove

File: demos/tests/cases.py [hunk 2]
```diff
@@ -310,33 +309,77 @@ def single_option_cases(key, *args):
...
+            '-m_lm': ModelArg('facial-landmarks-35-adas-0002'),
+            '-m_es': ModelArg('open-closed-eye-0001'),
+        }),
+        single_option_cases(
+            '-m_fd',
+            ModelArg('face-detection-adas-0001'),
+            ModelArg('face-detection-retail-0004')),
+    )),
+
+    CppDemo(name='gesture_recognition_demo', implementation='cpp_gapi',
```
Comment (anzhella-pankratova) [inline demos/tests/cases.py:357]: The repetition, need to remove

File: demos/tests/cases.py [hunk 3]
```diff
@@ -310,33 +309,77 @@ def single_option_cases(key, *args):
...
+        [
+            TestCase(options={'-m_a': ModelArg('asl-recognition-0004'), '-c': str(OMZ_DIR / 'data/dataset_classes/msasl100.json')}),
+            TestCase(options={'-m_a': ModelArg('common-sign-language-0001'),
+                              '-c': str(OMZ_DIR / 'data/dataset_classes/jester27.json')}),
+            TestCase(options={'-m_a': ModelArg('common-sign-language-0002'),
+                              '-c': str(OMZ_DIR / 'data/dataset_classes/common_sign_language12.json')}),
+        ],
+    )),
+
+    CppDemo(name='face_detection_mtcnn_demo', implementation='cpp_gapi',
```
Comment (anzhella-pankratova) [inline demos/tests/cases.py:373]: The repetition, need to remove

File: demos/tests/cases.py [hunk 4]
```diff
@@ -389,10 +432,30 @@ def single_option_cases(key, *args):
...
                 '-m_em': ModelArg('emotions-recognition-retail-0003'),
-                # '-m_hp': ModelArg('head-pose-estimation-adas-0001'),  TODO wait for 77686: INT8: Function doesn't have output with name angle_r_fc
+                '-m_hp': ModelArg('head-pose-estimation-adas-0001'),
+                '-m_lm': ModelArg('facial-landmarks-35-adas-0002'),
+            }),
+            TestCase(options={'-m': ModelArg('face-detection-adas-0001')})
+        ]
+    )),
+
+    CppDemo(name='interactive_face_detection_demo', implementation='cpp_gapi',
```
Comment (anzhella-pankratova) [inline demos/tests/cases.py:445]: The repetition, need to remove

File: demos/tests/cases.py [hunk 5]
```diff
@@ -715,6 +774,26 @@ def single_option_cases(key, *args):
...
+            '-i': DataPatternArg('instance-segmentation'),
+        }),
+        single_option_cases('-m',
+            ModelArg('instance-segmentation-person-0007'),
+            ModelArg('instance-segmentation-security-0091')),
+    ))
+
+]
+
+PYTHON_DEMOS = [
```
Comment (anzhella-pankratova) [inline demos/tests/cases.py:796]:
```suggestion
```

File: demos/tests/cases.py [hunk 6]
```diff
@@ -1383,7 +1455,7 @@ def single_option_cases(key, *args):
         [
             *single_option_cases('-m_i',
                 ModelArg('instance-segmentation-security-0002'),
-                # ModelArg('instance-segmentation-security-0091'), # Slow model
+                ModelArg('instance-segmentation-security-0091'), # Slow model
```
Comment (anzhella-pankratova) [inline demos/tests/cases.py:1458]:
```suggestion
                # ModelArg('instance-segmentation-security-0091'), # Slow model
```

**Review by anzhella-pankratova — CHANGES_REQUESTED** [review]

**Review by anzhella-pankratova — APPROVED** [review]

File: demos/tests/cases.py [hunk 7]
```diff
@@ -723,6 +718,13 @@ def single_option_cases(key, *args):
         ]
     )),
 
+    CppDemo(name='noise_suppression_demo', device_keys=['-d'], test_cases=combine_cases(
+        TestCase(options={'-i': TestDataArg('how_are_you_doing.wav')}),
+        single_option_cases('-m',
+            ModelArg('noise-suppression-denseunet-ll-0001'),
+            ModelArg('noise-suppression-poconetlike-0001')),
+    )),
+
```
Comment (anzhella-pankratova) [inline demos/tests/cases.py:727]:
```suggestion
```

Comment (VoronovaIntern) [reply demos/tests/cases.py]: Done

**Comment by VoronovaIntern** [comment]
Problem demos and models:

- [ ] social_distance_demo ( person-reidentification-retail-0286 and 0288)
- [ ] action_recognition_demo ( action-recognition-0001) monitors_extension wasn't found
- [ ] human_pose_estimation_3d_demo  Module 'pose_extractor' not found
- [ ] speech_recognition_deepspeech_demo No module named 'ctcdecode_numpy'
- [ ] whiteboard_inpainting_demo ( instance-segmentation-security-0091) Slow model

**Pull request merged**